src/nvim/api/vim.c

```diff
@@ -761,7 +761,7 @@ void nvim_set_vvar(String name, Object value, Error *err)
 ///          - kind: Set the |ui-messages| kind with which this message will be emitted.
 ///          - verbose: Message is controlled by the 'verbose' option. Nvim invoked with `-V3log`
 ///            will write the message to the "log" file instead of standard output.
-void nvim_echo(Array chunks, Boolean history, Dict(echo_opts) *opts, Error *err)
+void nvim_echo(ArrayOf(ArrayOf(String, 2)) chunks, Boolean history, Dict(echo_opts) *opts, Error *err)
```

Comment:
This could also be `ArrayOf(Tuple(String, String))`.

Reply:
`ArrayOf(Tuple(String, Union(String, Integer)))` would be more correct (in case that's supported).